core/src/rpc_service.rs

```diff
@@ -358,6 +358,20 @@ impl JsonRpcService {
 
         let ledger_path = ledger_path.to_path_buf();
 
+        // sadly, some parts of our current rpc implemention block the jsonrpc's
+        // _socket-listening_ event loop for too long, due to (blocking) long IO or intesive CPU,
+        // causing no further processing of incoming requests and ultimatily innocent clients timing-out.
+        // So create a (shared) multi-threaded event_loop for jsonrpc and set its .threads() to 1,
+        // so that we avoid the single-threaded event loops from being created automatically by
```

Comment:
@mvines fyi, this behavior (well, a bit odd default behavior) is unchanged even at the tip of the crate's master branch.

https://github.com/paritytech/jsonrpc/blob/91cf903aa8a083ebf05018952c33a961ba590c4f/server-utils/src/reactor.rs#L100
```rust
 		tb.core_threads(1);
```


https://github.com/paritytech/jsonrpc/blob/91cf903aa8a083ebf05018952c33a961ba590c4f/server-utils/src/reactor.rs#L37
```rust
 			UninitializedExecutor::Unspawned => RpcEventLoop::with_name(Some(name.into())).map(Executor::Spawned),
```
https://github.com/paritytech/jsonrpc/blob/91cf903aa8a083ebf05018952c33a961ba590c4f/http/src/lib.rs#L461-L466:

```rust
		let handles = (0..self.threads - 1)
			.map(|i| {
				let (local_addr_tx, local_addr_rx) = mpsc::channel();
				let (close, shutdown_signal) = oneshot::channel();
				let (done_tx, done_rx) = oneshot::channel();
				let eloop = UninitializedExecutor::Unspawned.init_with_name(format!("http.worker{}", i + 1))?;
				serve(
```